sync.go

```diff
@@ -205,17 +205,25 @@ func (s *Syncer) runLeader(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
+	var srcTargets []*Target
+
+	// Check for destination changes every 15 minutes
+	// (timer initialized to inf to ensure source gets initialized first)
+	d := time.Minute * 15
+	t := time.NewTimer(time.Second * (1 << 32))
+	defer t.Stop()
 
 	// Wait for an update, if we get one sync it
 	for {
 		logrus.Debugf("Waiting for targets from source")
-		var srcTargets []*Target
 		select {
 		case <-ctx.Done():
 			return ctx.Err()
 		case srcTargets = <-srcCh:
+			logrus.Debugf("Received targets from source: %+#v", srcTargets)
+		case <-t.C:
 		}
-		logrus.Debugf("Received targets from source: %+#v", srcTargets)
+		t.Reset(d)
```

Comment:
Since reset is being called regardless of channel pull you will need to call Stop() before and clear the channel as described in the docs (https://golang.org/pkg/time/#Timer.Reset); In addition make sure the channel clear after the stop is a non-blocking select to avoid deadlock.

Reply:
Done.